src/actions/tamanu/prepare_upgrade.rs

```diff
@@ -0,0 +1,160 @@
+use std::{
+	collections::HashSet,
+	fs,
+	path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use itertools::Itertools;
+use miette::{bail, IntoDiagnostic, Result, WrapErr};
+use node_semver::Version;
+use regex::Regex;
+use tracing::info;
+
+use crate::actions::{
+	tamanu::{
+		download::{download, make_url},
+		find_package,
+	},
+	Context,
+};
+
+use super::{download::ServerKind, find_existing_version, find_tamanu, TamanuArgs};
```

Comment:
```suggestion
use crate::actions::{
	tamanu::{
		download::{download, make_url, ServerKind},
		find_existing_version,
		find_package,
		find_tamanu,
		TamanuArgs,
	},
	Context,
};
```

Tooling (like rust-analyzer) is a bit annoying at this but here `super` is `crates::actions::tamanu`, so having both the super and the crate in use can be confusing.

Reply:
Oh no. I was pseudo aware of this and still getting through... Anyways, now I reviewed this again. Modules like `find`, `config` and `download` use `super` for `crate::action::tamanu` alongside `crate::actions`. (In `psql`, I introduced this inconsistency in this PR.) For consistency, should I avoid `super` in the `tamanu` module? Using only `crate` references would be easier for tooling at least.

Reply:
Using both super and crate isn't inherently an issue, it's when they overlap in what their effective scope is. In some cases it might be that super is shorter or clearer.

Reply:
Umm. I'll go with the option existing modules picked so I made `prepare_upgrade` consistent with them. There isn't a particular reason but I think it looks good.